.github/workflows/update-test-rules.yml

```diff
@@ -130,9 +148,33 @@ jobs:
               continue
             fi
             
-            cp source/$file destination/$file
-            yq -i '.testing_pr = env(pr_num)' destination/$file
-            yq -i '.testing_sha = env(sha)' destination/$file
+            # Extract the base filename
+            base_filename=$(basename "$file")
+            # Create the prefixed filename with PR number
+            prefixed_filename="${pr_num}_${base_filename}"
+            # Get the directory of the original file
+            dir_path=$(dirname "$file")
+            
+            # Ensure the destination directory exists
+            mkdir -p "destination/$dir_path"
+            
+            # Copy the file to destination with original path structure
+            cp "source/$file" "destination/$dir_path/${prefixed_filename}"
+            
+            # Generate a deterministic UUID based on the prefixed filename
+            # test-rules requires each rule to have a unique, but deterministic
+            # id.  This will be different than the auto generated one with the main branch
+            new_uuid=$(generate_uuid "${prefixed_filename}")
+            echo "Generated UUID for ${prefixed_filename}: ${new_uuid}"
+            
+            # Update the ID in the YAML file using yq
+            yq -i '.id = "'$new_uuid'"' "destination/$dir_path/${prefixed_filename}"
```

Comment:
Maybe there's some value in including the original `id` as another field like `original_id`. That would just make investigations a little easier if the name is being changed at the same time.

Reply:
Good suggestion! Got this added!